src/app/stack/card-details/card-details.component.ts

```diff
@@ -1018,13 +1018,29 @@ export class CardDetailsComponent implements OnChanges {
         return null;
     }
 
+    private getBgColorofProgressBar(severity: string) {
+        switch (severity) {
+            case 'critical':
+                return '#800101'
+            case 'high':
+                return '#fb3333'
+            case 'medium':
+                return '#ffa500'
+            case 'low':
+                return '#ffd900'
+            default:
+                break;
+        }
+    }
```

Comment:
Can't we express this as css class & just assign the class to element as `severity`?

Reply:
Ya, we can do it.

Reply:
@arajkumar I tried on multiple ways to archive it but it did not work out. The main reason is, the element is not in the template used to view the background color of progressBar, it exists in the template of component-snippet, whereas we are setting the value in card-details.component .

If you have some other approach, please share :)

I corrected the width issue of the arrow {^, >} You can try here: 
http://stack-report-ui-v2-shumathu-fabric8-analytics.devtools-dev.ext.devshift.net/#/analyze/75aa1be2295c4cbf8cf0769fab86efee?interframe=true&api_data=%7B%22access_token%22%3A%22undefined%22%2C%22route_config%22%3A%7B%22api_url%22%3A%22https%3A%2F%2Ff8a-analytics-preview-2445582058137.staging.gw.apicast.io%22%2C%22ver%22%3A%22v2%22%7D%2C%22user_key%22%3A%223e42fa66f65124e6b1266a23431e3d08%22%7D

Reply:
```
                        <div *ngSwitchCase="'highestCVSS'">
                            <!-- highestCVSS -->
                            <ana-progress-meter
                                *ngIf="comp?.privateSecurityDetails?.progressReport && tabType == 'private'"
                                [config]="comp?.privateSecurityDetails?.progressReport "></ana-progress-meter>
                            <ana-progress-meter
                                *ngIf="comp?.publicSecurityDetails?.progressReport && tabType == 'public'"
                                [config]="comp?.publicSecurityDetails?.progressReport "></ana-progress-meter>
                            <span *ngIf="tabType == 'public' && !comp?.publicSecurityDetails?.progressReport"
                                [innerText]="'-'"></span>
                            <span *ngIf="tabType == 'private' && !comp?.privateSecurityDetails?.progressReport"
                                [innerText]="'-'"></span>
                        </div>
```
The above snippet is what shows the progress meter right.? If it is the case, why can't add another property to `publicSecurityDetails, privateSecurityDetails` to carry the severity which can define the color of it through css class?

Reply:
Okay, now I got it. Yes We can do it by passing it as a new property, but the only issue I see is we may need to change or have multiple progress meters implementations one for security tab and other for Add-ons(confidence score).

![image](https://user-images.githubusercontent.com/23407383/86463937-1a063500-bd4c-11ea-83f1-2275055a15c3.png)

Reply:
if current progress meter takes only the percent, text and color, then keep as it is.

Reply:
Okay